includes/restapi/class-notification-controller.php

```diff
@@ -41,27 +42,36 @@ public function __construct() {
 	 *
 	 * @return void
 	 */
-	public function register_routes() : void {
+	public function register_routes(): void {
 		register_rest_route(
 			self::NAMESPACE,
 			self::NOTIFICATION_BASE,
 			array(
-				'methods'             => WP_REST_Server::READABLE,
-				'args'                => array(),
-				'callback'            => array( $this, 'get_notifications' ),
-				'permission_callback' => array( $this, 'authenticate' ),
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'args'                => $this->get_collection_params(),
+					'callback'            => array( $this, 'get_items' ),
+					'permission_callback' => array( $this, 'get_items_permissions_check' ),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
 			),
 			false
 		);
 	}
 
 	/**
-	 * Authenticate notification request.
-	 *
-	 * @return boolean true if authenticated, false otherwise.
+	 * Checks if a given request has access to read notifications.
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return true|WP_Error True if the request has access to view the items, error object otherwise.
 	 */
-	public function authenticate() : bool {
-		// TODO: Implement authentication.
+	public function get_items_permissions_check( $request ) {
```

Comment:
@Sephsekla This an `get_notifications` where renamed, because `get_items` and `get_items_permission_check` are standard names used in core rest-api.